next.mdx.shiki.mjs

```diff
@@ -19,8 +19,150 @@ const memoizedShikiji = await getHighlighterCore({
 // to attribute the current language of the <pre> element
 const languagePrefix = 'language-';
 
+/**
+ * Retrieve the value for the given meta key.
+ *
+ * @example - Returns "CommonJS"
+ * getMetaParameter('displayName="CommonJS"', 'displayName');
+ *
+ * @param {any} meta - The meta parameter.
+ * @param {string} key - The key to retrieve the value.
+ *
+ * @return {string | undefined} - The value related to the given key.
+ */
+function getMetaParameter(meta, key) {
+  if (typeof meta !== 'string') {
+    return;
+  }
+
+  const matches = meta.match(new RegExp(`${key}="(?<parameter>[^"]*)"`));
+  const parameter = matches?.groups.parameter;
+
+  return parameter !== undefined && parameter.length > 0
+    ? parameter
+    : undefined;
+}
+
+/**
+ * @typedef {import('unist').Node} Node
+ * @property {string} tagName
+ * @property {Node[]} children
+ */
+
+/**
+ * Checks if the given node is a valid code element.
+ *
+ * @param {Node} node - The node to be verified.
+ *
+ * @return {boolean} - True when it is a valid code element, false otherwise.
+ */
+function isCodeBlock(node) {
+  return node?.tagName === 'pre' && node?.children[0].tagName === 'code';
+}
+
+/**
+ * Retrieves a list indicating the starting, and ending indexes of sequential
+ * code elements.
+ *
+ * @param {Node} tree - The current MDX resolved content.
+ *
+ * @return {{start: number, end: number}[]} - The list containing every range of
+ * sequential code elements.
+ */
+function getCodeTabsRange(tree) {
+  const rangeMap = {};
+  let start = null;
+
+  visit(tree, 'element', (node, index, parent) => {
+    // Adding 2 since there is one text node between every element
+    const next = index + 2;
+
+    if (
```

Comment:
Can you separate this big if into maybe smaller if statements as I've done in the same file? And explain why these if statements exist and why we check for such things?